presto-hive/src/test/java/com/facebook/presto/hive/TestHiveDistributedQueries.java

```diff
@@ -55,4 +59,29 @@ public void testRenameTable()
     {
         // Hive connector currently does not support table rename
     }
+
+    @Test
+    public void testShowCreateTable()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_show_create_table (c1 bigint, c2 double, \"c 3\" varchar, \"c'4\" ARRAY <bigint>, c5 MAP <bigint, varchar>)");
+        String expectedShowCreateTableSql = format("CREATE TABLE %s.%s.%s (\n" +
+                        "  c1 bigint,\n  c2 double,\n  \"c 3\" varchar,\n  \"c'4\" array(bigint),\n  c5 map(bigint,varchar)\n)\n" +
+                        "WITH (\n  format=RCBINARY\n)",
+                getSession().getCatalog().get(), getSession().getSchema().get(), "test_show_create_table");
+        MaterializedResult actualResult = computeActual("SHOW CREATE TABLE test_show_create_table");
+        assertEquals(actualResult.getRowCount(), 1);
+        assertEquals(actualResult.getMaterializedRows().get(0).getFieldCount(), 1);
+        assertEquals(actualResult.getMaterializedRows().get(0).getField(0).toString(), expectedShowCreateTableSql);
+
+        assertUpdate("CREATE TABLE test_show_create_table2 (c1 bigint, \"c 2\" varchar, \"c'3\" ARRAY <bigint>, c4 MAP <bigint, varchar>, c5 double) WITH (format='ORC', partitioned_by = ARRAY['c5'])");
+        expectedShowCreateTableSql = format("CREATE TABLE %s.%s.%s (\n" +
+                        "  c1 bigint,\n  \"c 2\" varchar,\n  \"c'3\" array(bigint),\n  c4 map(bigint,varchar),\n  c5 double\n)\n" +
+                        "WITH (\n  format=ORC,\n  partitioned_by=[c5]\n)",
+                getSession().getCatalog().get(), getSession().getSchema().get(), "test_show_create_table2");
+        actualResult = computeActual("SHOW CREATE TABLE test_show_create_table2");
+        assertEquals(actualResult.getRowCount(), 1);
```

Comment:
`MaterializedResult` has a helper method `getOnlyColumnAsSet()`, so you can replace these three lines